build_runner/lib/src/asset_graph/node.dart

```diff
@@ -91,12 +92,22 @@ class GeneratedAssetNode extends AssetNode {
   /// this node.
   final AssetId builderOptionsId;
 
-  GeneratedAssetNode(this.phaseNumber, this.primaryInput, this.needsUpdate,
-      this.wasOutput, AssetId id, this.builderOptionsId,
-      {Digest lastKnownDigest,
-      Set<Glob> globs,
-      Iterable<AssetId> inputs,
-      this.previousInputsDigest})
+  /// Whether the asset should be placed in the build cache.
+  final bool isHidden;
+
+  GeneratedAssetNode(
+    this.phaseNumber,
+    this.primaryInput,
+    this.needsUpdate,
+    this.wasOutput,
+    AssetId id,
+    this.builderOptionsId, {
+    Digest lastKnownDigest,
+    Set<Glob> globs,
+    Iterable<AssetId> inputs,
+    this.previousInputsDigest,
+    @required this.isHidden,
```

Comment:
nit: I would just move this up with the other required args - if we want to go with @required we should do it for pretty much all the fields at once imo (same elsewhere)

Reply:
non-named boolean args are tough at the call sites.

Not all the named args here should be required but I can go through and find the ones that should be and mark them - and see if any of the non-named args should become named.
Can I follow up with that in a separate PR?

Reply:
https://github.com/dart-lang/build/pull/695

If you feel it should go out together I can merge it in here.

Reply:
Whichever is easiest